File: src/cpp/swap_nodes_pairs.cpp
/*
algos by Alfonso Gonzalez
https://github.com/alfonsogonzalez/algos
https://www.youtube.com/c/AlfonsoGonzalezSpaceEngineering

Implementation of swap_nodes_pairs function
*/
#include "swap_nodes_pairs.hpp"

#include <bits/stdc++.h>  // for std::swap function

ListNode* swap_nodes_pairs( ListNode* head ) {
  if ( head && head->next ) {
    std::swap( head->val, head->next->val );
    swap_nodes_pairs( head->next->next );
  }
  return head;
}

File: src/py/swap_nodes_pairs.py
'''
algos by Alfonso Gonzalez
https://github.com/alfonsogonzalez/algos
https://www.youtube.com/c/AlfonsoGonzalezSpaceEngineering

Implementation of swap_nodes_pairs function
'''

def swap_nodes_pairs( head ):
	if head and head.next:
		head.val, head.next.val = head.next.val, head.val
		swap_nodes_pairs( head.next.next )
	return head
